pkg/controller/actuator.go

```diff
@@ -483,3 +567,39 @@ func HashData(data interface{}) (string, error) {
 	bytes := sha256.Sum256(jsonSpec)
 	return strings.ToLower(base32.StdEncoding.EncodeToString(bytes[:]))[:16], nil
 }
+
+// findIstioNamepsaceForExtension finds the Istio namespace by the Istio Gateway
+// object named "kube-apiserver", which is expected to be present in every
+// Shoot namespace. This Gateway object has a Selector field that selects a
+// Deployment in the namespace we need. We list Deployments filtered by the
+// labelSelector and return the namespace of the returned Deployment.
+func (a *actuator) findIstioNamespaceForExtension(
+	ctx context.Context, ex *extensionsv1alpha1.Extension,
+) (
+	istioNamespace string,
+	istioLabels map[string]string,
+	err error,
+) {
+	gw := istionetworkv1beta1.Gateway{}
+
+	err = a.client.Get(ctx, client.ObjectKey{
+		Namespace: ex.Namespace,
+		Name:      istioGatewayName,
+	}, &gw)
+	if err != nil {
+		return "", nil, err
+	}
+
+	labelsSelector := client.MatchingLabels(gw.Spec.Selector)
+
+	deployments := appsv1.DeploymentList{}
+	err = a.client.List(ctx, &deployments, labelsSelector)
+	if err != nil {
+		return "", nil, err
+	}
+	if len(deployments.Items) != 1 {
+		return "", nil, fmt.Errorf("no istio namespace could be selected, because the numer of deployments found is %d", len(deployments.Items))
```

Comment:
```suggestion
		return "", nil, fmt.Errorf("no istio namespace could be selected, because the number of deployments found is %d", len(deployments.Items))
```